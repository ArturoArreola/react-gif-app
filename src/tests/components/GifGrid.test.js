import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid/>', () => { 

    const category = 'Hello';
    
    test('Component must show correctly', () => { 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
     });

     test('Component must show items when gifs are loaded throug useFetchGifs ', () => {
        const gifs = [{
            id: '123',
            url: 'http://localhost/image.png',
            title: 'My Gif'
        },
        {
            id: '456',
            url: 'http://localhost/image.png',
            title: 'My Gif'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(2);
     });
});