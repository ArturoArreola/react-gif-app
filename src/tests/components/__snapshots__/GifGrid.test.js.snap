// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid/> Component must show correctly 1`] = `
<Fragment>
  <h3>
    Hello
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
